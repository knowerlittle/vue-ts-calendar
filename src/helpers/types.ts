export type SelectedDate = {
  fromDate: string
  toDate: string
  label: string
}
