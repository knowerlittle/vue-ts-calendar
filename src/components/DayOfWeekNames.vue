<template>
  <div :class="[$style.weekday, { [$style.sticky]: sticky }]">
    <p
      v-for="dayName in dayNames"
      :key="dayName"
      :class="$style['weekday__names']"
    >
      {{ dayName }}
    </p>
  </div>
</template>

<script lang="ts">
import { Component, Vue, Prop } from 'vue-property-decorator'

@Component({ name: 'DayOfWeekNames' })
export default class DayOfWeekNames extends Vue {
  /**
   * Name of days
   */
  @Prop({ required: true })
  readonly dayNames!: string[]

  /**
   * Make the names sticky
   */
  @Prop({ default: false })
  readonly sticky!: boolean

  $style: any
}
</script>

<style lang="scss" module>
  .weekday {
    padding-top: 10px;
    background-color: white;
    display: grid;
    grid-template-columns: repeat(7, 1fr);

    &.sticky {
      position: sticky;
      top: 0;
      z-index: 1000;
    }

    &__names {
      color: gray;
      justify-self: center;
      padding: 4px 0;
    }
  }

</style>
