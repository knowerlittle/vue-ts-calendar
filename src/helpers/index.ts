export const monthNamesFallback = [
  'January',
  'February',
  'March',
  'April',
  'May',
  'June',
  'July',
  'August',
  'September',
  'October',
  'November',
  'December'
]

export const dayNamesFallback = [
  'Mon',
  'Tues',
  'Wed',
  'Thu',
  'Fri',
  'Sat',
  'Sun'
]

export enum ViewType {
  Single = 'single',
  Dual = 'dual',
  Infinite = 'infinite'
}
