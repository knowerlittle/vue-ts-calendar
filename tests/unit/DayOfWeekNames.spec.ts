import { shallowMount } from '@vue/test-utils'
import { dayNamesFallback } from '@/helpers'
import DayOfWeekNames from '@/components/DayOfWeekNames.vue'

describe('DayOfWeekNames', () => {
  const factory = (opts = {}) => shallowMount(DayOfWeekNames, {
    ...opts
  })

  describe('matches snapshot', () => {
    it('matches when header is not sticky', () => {
      const options = {
        propsData: {
          sticky: false,
          dayNames: dayNamesFallback
        }
      }
      const wrapper = factory(options)

      expect(wrapper.html()).toMatchSnapshot()
    })

    it('matches when header is sticky', () => {
      const options = {
        propsData: {
          sticky: true,
          dayNames: dayNamesFallback
        }
      }
      const wrapper = factory(options)

      expect(wrapper.html()).toMatchSnapshot()
    })
  })
})
