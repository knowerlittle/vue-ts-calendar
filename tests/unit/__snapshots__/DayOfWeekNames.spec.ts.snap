// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DayOfWeekNames matches snapshot matches when header is not sticky 1`] = `
<div class="weekday">
  <p tabindex="0" class="weekday__names">
    Mon
  </p>
  <p tabindex="0" class="weekday__names">
    Tues
  </p>
  <p tabindex="0" class="weekday__names">
    Wed
  </p>
  <p tabindex="0" class="weekday__names">
    Thu
  </p>
  <p tabindex="0" class="weekday__names">
    Fri
  </p>
  <p tabindex="0" class="weekday__names">
    Sat
  </p>
  <p tabindex="0" class="weekday__names">
    Sun
  </p>
</div>
`;

exports[`DayOfWeekNames matches snapshot matches when header is sticky 1`] = `
<div class="weekday sticky">
  <p tabindex="0" class="weekday__names">
    Mon
  </p>
  <p tabindex="0" class="weekday__names">
    Tues
  </p>
  <p tabindex="0" class="weekday__names">
    Wed
  </p>
  <p tabindex="0" class="weekday__names">
    Thu
  </p>
  <p tabindex="0" class="weekday__names">
    Fri
  </p>
  <p tabindex="0" class="weekday__names">
    Sat
  </p>
  <p tabindex="0" class="weekday__names">
    Sun
  </p>
</div>
`;
